src/libs/Navigation/linkTo.ts

```diff
@@ -152,16 +152,15 @@ export default function linkTo(navigation: NavigationContainerRef<RootStackParam
         const topRouteName = rootState?.routes?.at(-1)?.name;
         const isTargetNavigatorOnTop = topRouteName === action.payload.name;
 
+        const isTargetScreenDifferentThanCurrent = Boolean(topmostCentralPaneRoute && topmostCentralPaneRoute.name !== action.payload.params?.screen);
+        const areParamsDifferent = !shallowCompare(topmostCentralPaneRoute?.params, action.payload.params?.params);
```

Comment:
This also caused https://github.com/Expensify/App/issues/40743

Reply:
[The fix](https://github.com/Expensify/App/pull/41967/files) for that has been already merged. Could you check if it still occurs in dev environment?